src/CloudFoundry.VisualStudio/MSBuild/MSBuildProcess.cs

```diff
@@ -41,6 +51,19 @@ public static void Publish(EnvDTE.Project project, Dictionary<string, string> bu
                     pane = output.OutputWindowPanes.Add("Publish");
                 }
 
+                EnvDTE.Properties projectsAndSolution = dte.get_Properties("Environment", "ProjectsAndSolution");
```

Comment:
Can you extract this subroutine and put it in VSUtils.GetVisualStudioSetting(string a, string b, string c)?